tests/test_babis.py

```diff
@@ -9,21 +9,79 @@
 """
 
 import pytest
+from mock import ANY, Mock, call, patch
+from six.moves.urllib.error import URLError
 
 
-from babis import babis
+from babis import decorator
 
 
 class TestBabis(object):
+    def test_before_and_after(self):
+        first_request = Mock()
+        second_request = Mock()
+        third_request = Mock()
 
-    @classmethod
-    def setup_class(cls):
-        pass
+        with patch('babis.babis.Request') as Request:
+            Request.side_effect = [
+                first_request,
+                third_request
+            ]
+            with patch('babis.babis.urlopen') as urlopen:
+                def wrapped():
+                    urlopen(second_request)
+                decorator(ping_before='foo', ping_after='bar')(wrapped)()
 
-    def test_something(self):
-        pass
+        urlopen.assert_has_calls([
+            call(first_request), call(second_request), call(third_request)])
 
-    @classmethod
-    def teardown_class(cls):
-        pass
+    def test_ping_urls_string(self):
+        dec = decorator()
+        dec.ping_url = Mock()
+        dec.ping_urls('foo')
+        dec.ping_url.assert_called_with('foo')
 
+    def test_ping_urls_iterable(self):
+        dec = decorator()
+        dec.ping_url = Mock()
+        dec.ping_urls(('foo', 'bar'))
+        dec.ping_url.assert_has_calls([call('foo'), call('bar')])
+
+    def test_ping_url(self):
+        dec = decorator()
+        with patch('babis.babis.Request') as Request:
+            request_init_mock = Mock()
+            Request.return_value = request_init_mock
+            with patch('babis.babis.urlopen') as urlopen:
+                dec.ping_url('foo')
+        Request.assert_called_with('foo', data=ANY, headers=ANY)
```

Comment:
Since ``babis`` is doing multiple ``urlopen`` calls I would explicitly check ``assert_called_once_with``.